feature/google/src/main/java/com/revenuecat/purchases/google/findBestOffer.kt

```diff
@@ -0,0 +1,62 @@
+@file:Suppress("MaximumLineLength", "MaxLineLength")
+package com.revenuecat.purchases.google
+
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.models.PricingPhase
+
+fun List<GoogleSubscriptionOption>.findBestOffer(): GoogleSubscriptionOption? {
+    val basePlan = this.firstOrNull { it.isBasePlan } ?: return null
+
+    val validOffers = this
+        .filter { !it.isBasePlan }
+        .filter { !it.tags.contains("rc-ignore-best-offer") }
+
+    return findLongestFreeTrial(validOffers) ?: findLowestNoneFreeOffer(validOffers) ?: basePlan
+}
+
+private fun findLongestFreeTrial(offers: List<GoogleSubscriptionOption>): GoogleSubscriptionOption? {
+    return offers.mapNotNull { offer ->
+        offer.freePricingPhase()?.let { pricingPhase ->
+            Pair(offer, parseBillPeriodToDays(pricingPhase.billingPeriod))
+        }
+    }.maxByOrNull { it.second }?.first
+}
+
+private fun findLowestNoneFreeOffer(offers: List<GoogleSubscriptionOption>): GoogleSubscriptionOption? {
+    return offers.mapNotNull { offer ->
+        offer.nonFreePricingPhase()?.let { pricingPhase ->
+            Pair(offer, pricingPhase.priceAmountMicros)
+        }
+    }.minByOrNull { it.second }?.first
+}
+
+private fun GoogleSubscriptionOption.freePricingPhase(): PricingPhase? {
```

Comment:
not a huge diff, but these could be computed properties instead..

Reply:
Ha! I tried that but couldn't figure out the syntax 🤦‍♂️ Will try again!

Reply:
check out `SubscriptionOfferDetails.subscriptionBillingPeriod` for an example!